src/gen/local_lm.py

```diff
@@ -15,14 +19,19 @@ def get_model(model_id, model_revision, load_in_8bit, load_in_4bit, model_args,
     """
     model_id = sft_args.hub_model_id if model_id is None else model_id
     tokenizer = get_tokenizer(model_args, data_args)
-    quantization_config = BitsAndBytesConfig(load_in_8bit=load_in_8bit, load_in_4bit=load_in_4bit)
-    
-    model = AutoModelForCausalLM.from_pretrained(
-        model_id,
-        revision=model_revision, 
-        quantization_config=quantization_config, 
-        torch_dtype=torch.bfloat16, device_map="auto"
-    )
+    torch_dtype = (
+        model_args.torch_dtype if model_args.torch_dtype in ["auto", None] else getattr(torch, model_args.torch_dtype)
+    )    
+    quantization_config = get_quantization_config(model_args)
+    model_kwargs = dict(
+        revision=model_revision,
+        trust_remote_code=model_args.trust_remote_code,
+        use_flash_attention_2=model_args.use_flash_attention_2,
+        torch_dtype=torch_dtype,
+        device_map=get_kbit_device_map() if quantization_config is not None else None,
```

Comment:
Even when quantization_config is None user may want to set this to auto to run on GPUs that don't have huge VRAMs.

Reply:
then just "auto" would be fine?

Reply:
Having "auto" default sounds reasonable to me. Or we could let the user pass it.

Reply:
done! "auto" for now